### ISSUE TITLE
Include Realm in Eclipse library 

### ISSUE THREAD

**Issue by batista-mihai** [realm/realm-java#1475]
Hello,

I have some troubles including Realm in Eclipse library, the issues is somehow similar with the one reported here https://github.com/realm/realm-java/issues/817 , but the solution provided here didn't helped me solve my issue.
Environment configuration: Eclipse Luna (4.4.2) / JDK 1.7 / API 22 / Realm 0.82.2

The project structure is the following: The code that uses realm is packed in a jar file, this jar is included in a Android library project that has in the libs the realm jar and the folders with .so files, this project has the annotations processor setup. This Android library is included in a project.

In the jar, the classes extending RealmObject have the @RealmClass annotation.

If I create a custom module to have a different schema for the library. Module defined as below:
```java
@RealmModule(library = true, classes = {NameOfRealmModel1.class, NameOfRealmModel2.class } or allClasses = true)
public class RLibraryModule {
}
```
The following exception is thrown
```java
09-16 20:18:14.005: W/System.err(27890): io.realm.exceptions.RealmException: Could not find io.realm.RLibraryModuleMediator
09-16 20:18:14.005: W/System.err(27890): 	at io.realm.RealmConfiguration.getModuleMediator(RealmConfiguration.java:196)
09-16 20:18:14.005: W/System.err(27890): 	at io.realm.RealmConfiguration.createSchemaMediator(RealmConfiguration.java:173)
09-16 20:18:14.005: W/System.err(27890): 	at io.realm.RealmConfiguration.<init>(RealmConfiguration.java:88)
09-16 20:18:14.005: W/System.err(27890): 	at io.realm.RealmConfiguration.<init>(RealmConfiguration.java:54)
09-16 20:18:14.005: W/System.err(27890): 	at io.realm.RealmConfiguration$Builder.build(RealmConfiguration.java:415)
09-16 20:18:14.005: W/System.err(27890): 	at com.onyxbeacon.service.ContentManager.<init>(ContentManager.java:148)
09-16 20:18:14.005: W/System.err(27890): 	at com.onyxbeacon.service.ContentManager.getInstance(ContentManager.java:111)
09-16 20:18:14.005: W/System.err(27890): 	at com.onyxbeacon.service.OnyxBeaconService.serviceSetup(OnyxBeaconService.java:446)
09-16 20:18:14.005: W/System.err(27890): 	at com.onyxbeacon.service.OnyxBeaconService.onStartCommand(OnyxBeaconService.java:337)
09-16 20:18:14.005: W/System.err(27890): 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2869)
09-16 20:18:14.005: W/System.err(27890): 	at android.app.ActivityThread.access$2100(ActivityThread.java:139)
09-16 20:18:14.005: W/System.err(27890): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1390)
09-16 20:18:14.005: W/System.err(27890): 	at android.os.Handler.dispatchMessage(Handler.java:102)
09-16 20:18:14.005: W/System.err(27890): 	at android.os.Looper.loop(Looper.java:149)
09-16 20:18:14.005: W/System.err(27890): 	at android.app.ActivityThread.main(ActivityThread.java:5257)
09-16 20:18:14.015: W/System.err(27890): 	at java.lang.reflect.Method.invokeNative(Native Method)
09-16 20:18:14.015: W/System.err(27890): 	at java.lang.reflect.Method.invoke(Method.java:515)
09-16 20:18:14.015: W/System.err(27890): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
09-16 20:18:14.015: W/System.err(27890): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609)
09-16 20:18:14.015: W/System.err(27890): 	at dalvik.system.NativeStart.main(Native Method)
09-16 20:18:14.015: W/System.err(27890): Caused by: java.lang.ClassNotFoundException: io.realm.RLibraryModuleMediator
09-16 20:18:14.015: W/System.err(27890): 	at java.lang.Class.classForName(Native Method)
09-16 20:18:14.015: W/System.err(27890): 	at java.lang.Class.forName(Class.java:251)
09-16 20:18:14.015: W/System.err(27890): 	at java.lang.Class.forName(Class.java:216)
09-16 20:18:14.015: W/System.err(27890): 	at io.realm.RealmConfiguration.getModuleMediator(RealmConfiguration.java:191)
09-16 20:18:14.015: W/System.err(27890): 	... 19 more
09-16 20:18:14.015: W/System.err(27890): Caused by: java.lang.NoClassDefFoundError: io/realm/RLibraryModuleMediator
09-16 20:18:14.015: W/System.err(27890): 	... 23 more
09-16 20:18:14.015: W/System.err(27890): Caused by: java.lang.ClassNotFoundException: Didn't find class "io.realm.RLibraryModuleMediator" on path: DexPathList[[zip file "/data/app/com.test.realmtest-2.apk"],nativeLibraryDirectories=[/data/app-lib/com.test.realmtest-2, /vendor/lib, /system/lib, /system/lib/arm]]
09-16 20:18:14.025: W/System.err(27890): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
09-16 20:18:14.025: W/System.err(27890): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
09-16 20:18:14.025: W/System.err(27890): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
09-16 20:18:14.025: W/System.err(27890): 	... 23 more
```

In case I go with the default schema. RDevice is a realm model class that extends RealmObject. The following exception is thrown:
```java
09-16 20:35:17.385: W/System.err(28923): java.lang.IllegalArgumentException: RDevice is not part of the schema for this Realm
09-16 20:35:17.385: W/System.err(28923): 	at io.realm.internal.modules.CompositeMediator.getMediator(CompositeMediator.java:118)
09-16 20:35:17.385: W/System.err(28923): 	at io.realm.internal.modules.CompositeMediator.getTableName(CompositeMediator.java:71)
09-16 20:35:17.385: W/System.err(28923): 	at io.realm.Realm.getTable(Realm.java:329)
09-16 20:35:17.385: W/System.err(28923): 	at io.realm.RealmQuery.<init>(RealmQuery.java:75)
09-16 20:35:17.385: W/System.err(28923): 	at io.realm.Realm.where(Realm.java:1217)
09-16 20:35:17.385: W/System.err(28923): 	at com.mylib.service.account.AccountManager.getList(AccountManager.java:299)
09-16 20:35:17.385: W/System.err(28923): 	at com.mylib.service.account.AccountManager.initialize(AccountManager.java:119)
09-16 20:35:17.385: W/System.err(28923): 	at com.mylib.service.OnyxBeaconService.initialize(MyService.java:479)
09-16 20:35:17.385: W/System.err(28923): 	at com.mylib.service.OnyxBeaconService.onStartCommand(MyService.java:180)
09-16 20:35:17.385: W/System.err(28923): 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2869)
09-16 20:35:17.385: W/System.err(28923): 	at android.app.ActivityThread.access$2100(ActivityThread.java:139)
09-16 20:35:17.385: W/System.err(28923): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1390)
09-16 20:35:17.385: W/System.err(28923): 	at android.os.Handler.dispatchMessage(Handler.java:102)
09-16 20:35:17.385: W/System.err(28923): 	at android.os.Looper.loop(Looper.java:149)
09-16 20:35:17.385: W/System.err(28923): 	at android.app.ActivityThread.main(ActivityThread.java:5257)
09-16 20:35:17.395: W/System.err(28923): 	at java.lang.reflect.Method.invokeNative(Native Method)
09-16 20:35:17.395: W/System.err(28923): 	at java.lang.reflect.Method.invoke(Method.java:515)
09-16 20:35:17.395: W/System.err(28923): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
09-16 20:35:17.395: W/System.err(28923): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609)
09-16 20:35:17.395: W/System.err(28923): 	at dalvik.system.NativeStart.main(Native Method)
```
I can provide other details about the Realm implementations if necessary.
For the moment I'm stuck, any ideas about things that I could try would be helpful.

Thank you,
Mihai

**Comment by cmelchior** [comment]
If I understand you correctly you have the following setup:

```
------
Jar with Realm code
------
^
------
Library project with APT/Realm jar
------
^
------
App project
------
```

Is that correct?

**Comment by batista-mihai** [comment]
Is correct

**Comment by cmelchior** [comment]
It is the the project that have your Realm model classes that need the Annotation processor, so if I understand you correctly changing it to the following should work:

```
------
Jar with Realm models + APT/Realm jar
------
^
------
Library project 
------
^
------
App project
------
```

**Comment by batista-mihai** [comment]
I'll try and come back, I thought that the realm generated files won't be exported in the jar

**Comment by batista-mihai** [comment]
Thank you for the advice

**Comment by kneth** [comment]
@batista-mihai Did you find a solution?

**Comment by cmelchior** [comment]
Closing due to no response and question seems to be answered.

**Comment by batista-mihai** [comment]
Sorry for not replying in time, I write this for anyone interested about this scenario, this is what worked for me. 

The cmelchior reply was helpful, thank you. I configured the project that contains the Realm models and db operations with annotations. After build, I've exported a jar and included the .apt_generated folder. 
I've added the jar to the build path of the Library project and configured also the annotations for Realm.
